### PR TITLE
ui: Refactor the `ColorStackLegend` component to be outside the icicle graph component

### PR DESCRIPTION
**Pull request by yomete**
This ensures that the legend will always be persistent (i.e. the metadata call isn't called again) when clicking through binaries.

### CODE REVIEW

File: ui/packages/shared/profile/src/ProfileIcicleGraph/IcicleGraphArrow/index.tsx [hunk 1]
```diff
@@ -106,50 +126,55 @@ export const IcicleGraphArrow = memo(function IcicleGraphArrow({
 
   const currentSearchString = (selectQueryParam('search_string') as string) ?? '';
   const {compareMode} = useProfileViewContext();
-  const isColorStackLegendEnabled = selectQueryParam('color_stack_legend') === 'true';
+  // const isColorStackLegendEnabled = selectQueryParam('color_stack_legend') === 'true';
```
Comment (brancz) [inline ui/packages/shared/profile/src/ProfileIcicleGraph/IcicleGraphArrow/index.tsx:129]: perhaps we're ready to remove this setting?

File: ui/packages/shared/profile/src/ProfileIcicleGraph/index.tsx [hunk 2]
```diff
@@ -354,11 +345,68 @@ const ProfileIcicleGraph = function ProfileIcicleGraphNonMemo({
     return <ErrorContent errorMessage={capitalizeOnlyFirstLetter(error.message)} />;
   }
 
-  if (graph === undefined && arrow === undefined)
-    return <div className="mx-auto text-center">No data...</div>;
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const icicleGraph = useMemo(() => {
+    if (isLoading) {
```
Comment (brancz) [inline ui/packages/shared/profile/src/ProfileIcicleGraph/index.tsx:350]: we need to also block on `mappingsLoading` here, no?

File: ui/packages/shared/profile/src/ProfileIcicleGraph/IcicleGraphArrow/index.tsx (hunk 1 above)
Comment (yomete) [reply ui/packages/shared/profile/src/ProfileIcicleGraph/IcicleGraphArrow/index.tsx:129]: mean't to delete this, must have missed this.

File: ui/packages/shared/profile/src/ProfileIcicleGraph/index.tsx (hunk 2 above)
Comment (yomete) [reply ui/packages/shared/profile/src/ProfileIcicleGraph/index.tsx:350]: not really, `loading` in the `useEffect` a couple of lines above this code block is already a combination of `flamegraphLoading && profilemetadataLoading` from the parent component's parent component.

```
useEffect(() => {
    if (!loading && (arrow !== undefined || graph !== undefined)) {
      setIsLoading(false);
    } else {
      setIsLoading(true);
    }
  }, [loading, arrow, graph]);
```

**Comment by brancz** [comment]
Nevermind, discussed on discord:

1) loading already checks both https://github.com/parca-dev/parca/blob/8fc2a9919e1afcd7583b08d7757c68c770249bb4/ui/packages/shared/profile/src/ProfileViewWithData.tsx#L192
2) which is set using the combined loading and isloading https://github.com/parca-dev/parca/blob/8fc2a9919e1afcd7583b08d7757c68c770249bb4/ui/packages/shared/profile/src/ProfileIcicleGraph/index.tsx#L330-L336

**Pull request merged**